Python/ceval.c

```diff
@@ -439,35 +439,83 @@ int pthread_attr_destroy(pthread_attr_t *a)
 #endif
 
 
+static void
+tstate_set_stack(PyThreadState *tstate,
+                 void *stack_start_addr, size_t stack_size)
+{
+    assert(stack_size > 0);
+
+    _PyThreadStateImpl *ts = (_PyThreadStateImpl *)tstate;
+    ts->c_stack_hard_limit = (uintptr_t)stack_start_addr;
```

Comment:
This is incorrect. You need to leave `_PyOS_STACK_MARGIN_BYTES` space.

Don't worry, everyone seems to get this wrong. If https://github.com/python/cpython/pull/139294 ever gets reviewed and merged, it should prevent that happening.

Reply:
Fixed.